src/CommandLineOptions.ts

```diff
@@ -0,0 +1,51 @@
+import { Command } from 'commander'
+
+import packageJson from '../package.json'
+
+import * as lsif from './lsif'
+
+/** Configuration options to index a multi-project workspace. */
+export interface MultiProjectOptions {
+  inferTsconfig: boolean
+  progressBar: boolean
+  yarnWorkspaces: boolean
+  cwd: string
+  output: string
+  indexedProjects: Set<string>
+}
+
+/** Configuration options to index a single TypeScript project. */
+export interface ProjectOptions extends MultiProjectOptions {
+  projectRoot: string
+  projectDisplayName: string
+  writeIndex: (index: lsif.lib.codeintel.lsiftyped.Index) => void
+}
+
+export function commanderCommand(
+  indexAction: (projects: string[], otpions: MultiProjectOptions) => void
+): Command {
+  const command = new Command()
+  command
+    .name('lsif-typescript')
+    .version(packageJson.version)
+    .description('LSIF indexer for TypeScript and JavaScript')
+  command
+    .command('index')
+    .option('--cwd', 'the working directory', process.cwd())
+    .option('--yarn-workspaces', 'whether to index all yarn workspaces', false)
+    .option(
+      '--infer-tsconfig',
+      "whether to infer the tsconfig.json file, if it's missing",
+      false
+    )
+    .option('--output', 'path to the output file', 'dump.lsif-typed')
+    .option('--no-progress-bar', 'whether to disable the progress bar')
+    .argument('[projects...]')
```

Comment:
Now I'm wondering... is there a difference between:

```
lsif-typescript index --cwd /some/path
vs
lsif-typescript index /some/path
```

?

Reply:
It's good you asked because it surfaced a bug in how we parse the `--cwd` flag. I fixed the bug and added a cli test for `--cwd`. There is a difference between those two:

- `index --cwd /path` generates `/path/dump.lsif-typed`
- `index /path` generates `$PWD/dump.lsif-typed`